scylla/src/transport/session_test.rs

```diff
@@ -2885,3 +2885,36 @@ async fn test_manual_primary_key_computation() {
         .await;
     }
 }
+
+#[cfg(cassandra_tests)]
+#[tokio::test]
+async fn test_vector_type() {
+    setup_tracing();
+    let session = create_new_session_builder().build().await.unwrap();
+    let ks = unique_keyspace_name();
+
+    session.query(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1}}", ks), &[]).await.unwrap();
+    session
+        .query(
+            format!(
+                "CREATE TABLE IF NOT EXISTS {}.t (a int PRIMARY KEY, b vector<int, 4>, c vector<text, 2>)",
+                ks
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
+
+    session
+        .query(
+            format!(
+                "INSERT INTO {}.t (a, b, c) VALUES (1, [1, 2, 3, 4], ['foo', 'bar'])",
+                ks
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
```

Comment:
Important: missing test for a prepared statement combined with Vector.

Reply:
Question: is Vector insertable as bind value?
I.e., is the following query valid?
`INSERT INTO {}.t (a, b, c) VALUES (1, ?, ?)`
If so, `ColumnType::Vector` variant must be added and ser/de for it must be supported.

Reply:
I added a test using prepared statements.

I've added bind values to the TODO at the bottom of the test. Lets address them in the follow up work, since when we add `ColumnType::Vector` we will need to add support for all the other vector functionality at the same time.